### PR TITLE
Event Registration Cancellation #48

### LINKED ISSUE #48: Event Registration Cancellation
- I should be able to cancel an event registration. need to keep canceled records for reporting purposes

- When an event registration is canceled, session registrations under that event should also be canceled. 

- If there is no active event registration for an event, session registration can not be created under the sessions of the event

- CANCELED REQUIREMENT: Canceled event registrations should be excluded from overlap validations. Same way, if a canceled registration is activated, the overlap validations should be applied. INSTEAD OF OVERLAP, ANOTHER REQUIREMENT FOR DUPLICATE EVENT REGISTRATIONS: There can be multiple canceled event registrations and 1 active event registration on the system which belongs to a same event and contact 
- Event Registration Rollup Fields should be updated based on cancelation.

### PR DESCRIPTION
**Pull request by t-gul**
PR:

* Boolean field is created to control for active or not (cancelled) for event registration
* Boolean field is created to control for active or not (cancelled) for session registration
* Flow “After Insert Event Registration” status is updated with “created or updated”, I create a logic to cancel session registrations under cancelled event registrations for updates.
* New criteria is added into decision at flow “Before Save Session Registration Validations” to control event registration is cancelled or not. If it’s cancelled, “Is Event Registered" field value isn’t set true.
* Validation rule “Event_Not_Registered” under “Session Registration” object error message is updated.
* “Cancelled Text” text field under object "Event Registration” is created for matching rule of duplicate rule.  It is set with “Cancelled” boolean field value via flow “Create Event Registration Id & Fill Registration Fee”.
* “Registration Id” field’s unique checkbox is unchecked.
* New duplicate rule (it will check “Cancelled Text” and “Registration Id” fields) is created. Duplicate rule is active, If “Cancelled” field is false. (Attendee can have multiple session registration cancellations)
* Record page and page layout is updated

Test Cases Pdf :
[Test Cases#48.pdf](https://github.com/sgoEma/EventManagement/files/8531404/Test.Cases.48.pdf)

Org Credentials:
Username: test-ck9gyzzwcxtj@example.com
Password: -5evGoevfsorh

Note:
When test is doing, roll-up summary fields on “Public Event” object count or sum cancelled event registrations. Ticket can be create for this problem. FYI

### CODE REVIEW

**Review by sercanEmakina — COMMENTED** [review]
I believe we can figure another solution out later, but I think that's enough for now. well done!

**Review by bmtemakina — APPROVED** [review]

**Review by sercanEmakina — APPROVED** [review]

**Comment by sgoEma** [comment]
@t-gul please add duplicate and matching rules to the package.xml

File: force-app/main/default/profiles/Admin.profile-meta.xml [hunk 1]
```diff
@@ -237,7 +252,7 @@
     </flowAccesses>
     <flowAccesses>
         <enabled>false</enabled>
-        <flow>After_Insert_Event_Registration</flow>
```
Comment (sgoEma) [inline force-app/main/default/profiles/Admin.profile-meta.xml:240]: @t-gul should "after insert event registration" flow be removed?

Comment (t-gul) [reply force-app/main/default/profiles/Admin.profile-meta.xml:240]: @sgoEma Yes It's Changed with "After_Insert_Update_Event_Registration" flow.

**Comment by t-gul** [comment]
> @t-gul please add duplicate and matching rules to the package.xml

Package.xml is Updated with last commit.

**Comment by sercanEmakina** [comment]
@t-gul thank you for the test cases !

File: force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml [hunk 2]
```diff
@@ -3,8 +3,8 @@
     <actionCalls>
         <name>Event_Reg_Notification</name>
```
Comment (sgoEma) [inline force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml:4]: @t-gul I could not see any decision element on the flow which checks the "canceled" checkbox. It seems whenever an event registration is updated, the relevant session registrations will be canceled by the flow automatically. Is that correct?

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml [hunk 3]
```diff
@@ -1,12 +1,44 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Flow xmlns="http://soap.sforce.com/2006/04/metadata">
     <apiVersion>54.0</apiVersion>
+    <assignments>
```
Comment (sgoEma) [inline force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:4]: @t-gul On the flow, I see the first decision element named "Id_Check". However, the first outcome checks only "canceled" field.

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml [hunk 4]
```diff
@@ -1,12 +1,44 @@
...
+            <value>
+                <elementReference>$Record.Cancelled__c</elementReference>
+            </value>
+        </assignmentItems>
+        <connector>
+            <targetReference>Create_Unique_Id</targetReference>
+        </connector>
+    </assignments>
     <assignments>
         <description>Create Unique Id from Public Event Id &amp; Contact Id</description>
```
Comment (sgoEma) [inline force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:21]: @t-gul Even if only "cancelled" field is changed, you are populating the junction id

File: force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml (hunk 2 above)
Comment (t-gul) [reply force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml:4]: @sgoEma I will check it. If there is a problem like that, I will update flow

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml (hunk 3 above)
Comment (t-gul) [reply force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:4]: @sgoEma Yes, the flow work like that logic.

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml (hunk 4 above)
Comment (t-gul) [reply force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:21]: @sgoEma Yes, I can update this with new decision.

Thanks for review.

File: force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml (hunk 2 above)
Comment (t-gul) [reply force-app/main/default/flows/After_Insert_Update_Event_Registration.flow-meta.xml:4]: @sgoEma Flow is updated.

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml (hunk 3 above)
Comment (t-gul) [reply force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:4]: @sgoEma Flow is updated.

File: force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml (hunk 4 above)
Comment (t-gul) [reply force-app/main/default/flows/Create_Event_Registration_Id_Fill_Registration_Fee.flow-meta.xml:21]: @sgoEma Flow is updated.

**Review by sgoEma — APPROVED** [review]

**Pull request merged**